File: scripts/01_fetch_data.R
# 01_fetch_data.R ----------------------------------------------------------- #
# Author: Louis Trocellier
# Description: Script for step 01 - Fetching data.

# Source associated functions ----------------------------------------------- #
source(file.path(cfg$paths$functions, "01_fetch_data.R"), local = TRUE)

# Execution ----------------------------------------------------------------- #
log_info("Starting step 01: Fetch data")

# 1. Load parameters/settings
settings <- fetch_data_settings()

# 2. Retrieve data
raw_data <- fetch_data_execution(settings)

# 3. Export raw data to intermediate storage
fetch_data_exports(raw_data)

log_info("Step 01 completed successfully")

File: scripts/01_fetch_data.py
# scripts/01_fetch_data.py --------------------------------------------- #
# Author: Louis Trocellier
# Description: Pipeline step 01 - Fetch data execution.

from src.functions.f01_fetch_data import (
    fetch_data_settings,
    fetch_data_execution,
    fetch_data_exports
)

if __name__ == "__main__":
    # 1. Load step configuration/settings
    settings = fetch_data_settings()
    
    # 2. Retrieve raw data
    data = fetch_data_execution(settings)
    
    # 3. Export data in Parquet format
    fetch_data_exports(data)